timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/LockCollection.java

```diff
@@ -21,17 +21,18 @@
 
 final class LockCollection {
     private final ExclusiveLockCollection exclusiveLocks = new ExclusiveLockCollection();
-    private final ImmutableTimestampTracker immutableTimestampTracker = new ImmutableTimestampTracker();
+    private final NamedMinTimestampLockCollection namedMinTimestampLockCollection =
+            new NamedMinTimestampLockCollection();
 
     OrderedLocks getAllExclusiveLocks(Set<LockDescriptor> descriptors) {
         return exclusiveLocks.getAll(descriptors);
     }
 
     Optional<Long> getImmutableTimestamp() {
-        return immutableTimestampTracker.getImmutableTimestamp();
+        return namedMinTimestampLockCollection.getImmutableTimestamp();
     }
 
     AsyncLock getImmutableTimestampLock(long timestamp) {
-        return immutableTimestampTracker.getLockFor(timestamp);
+        return namedMinTimestampLockCollection.getImmutableTimestampLock(timestamp);
```

Comment:
See for what the full API will look like https://github.com/palantir/atlasdb/compare/aa-nmtl-trackers?expand=1.

I will have two exposed methods for generic timestamps. Those should only accept "CommitImmutableTimestamp" (and especially NOT ImmutableTimestamp)